src/node_platform.cc

```diff
@@ -0,0 +1,78 @@
+#include "node_platform.h"
+
+namespace node {
+
+using v8::Isolate;
+using v8::Platform;
+using v8::Task;
+using v8::TracingController;
+
+NodePlatform::NodePlatform(int thread_pool_size, uv_loop_t* loop,
+                           TracingController* tracing_controller)
+    : num_threads_(thread_pool_size), loop_(loop) {
+  if (tracing_controller) {
+    tracing_controller_.reset(tracing_controller);
+  } else {
+    TracingController* controller = new TracingController();
+    tracing_controller_.reset(controller);
+  }
+}
+
+NodePlatform::~NodePlatform() {}
+
+size_t NodePlatform::NumberOfAvailableBackgroundThreads() {
+  return static_cast<size_t>(num_threads_);
+}
+
+static void RunBackgroundTask(uv_work_t* handle) {
+  Task* task = static_cast<Task*>(handle->data);
+  task->Run();
+  delete task;
+}
+
+static void AfterBackgroundTask(uv_work_t* handle, int status) {
+  delete handle;
+}
+
+static void RunForegroundTask(uv_timer_t* handle) {
+  Task* task = static_cast<Task*>(handle->data);
+  task->Run();
+  delete task;
+  delete handle;
```

Comment:
This needs to call `uv_close()` and not free the memory until the close callback is called.